src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -999,17 +1003,31 @@ lit_expr ::= lit
 // Macros
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 
-macro_invocation ::= IDENTIFIER '!'
+fake macro ::= macro_invocation macro_arg?
 
-macro_expr ::= build_in_macro | macro_invocation macro_arg
+// Various kinds of macros
+macro_call ::= macro_invocation macro_arg
+item_macro ::= macro_invocation item_macro_arg
 
-private build_in_macro ::= try_macro_invocation '(' any_expr ')'
-                         | format_like_macro_invocation '(' format_macro_args ')'
+try_macro ::= try_macro_invocation try_macro_args { pin = 1 }
+format_like_macro ::= format_like_macro_invocation format_macro_args { pin = 1 }
 
-try_macro_invocation ::= "try" '!' {
-  elementType = macro_invocation
+macro_definition ::= macro_rules_invocation IDENTIFIER item_macro_arg {
+  pin = 1
+  extends = item_macro
 }
 
+private build_in_macro ::= try_macro
+                         | format_like_macro
+
+private macro_call_ ::= build_in_macro | macro_call { pin(".*") = macro_invocation }
+private macro_item_ ::= macro_definition | item_macro { pin(".*") = macro_invocation }
```

Comment:
I think we'd better avoid trailing `_` in rule names: https://github.com/JetBrains/Grammar-Kit/issues/97